<!DOCTYPE html>
<html>

  <head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	<title>## vcftools 的简单使用</title>

	<link rel="shortcut icon" href="/styles/images/favicon.jpg">
	<link rel="icon" href="/styles/images/favicon.jpg">

	<link rel="stylesheet" href="/styles/css/index.css">
	<link rel="stylesheet" href="/styles/css/fontawesome/css/font-awesome.min.css">
	<link rel="canonical" href="/2017/04/24/vcftools-introduction/">
	<link rel="alternate" type="application/rss+xml" title="Gossie's daily life" href="/feed.xml">
	
	<meta name="description" content="">

	<script src="/styles/js/jquery.min.js"></script>
	<!--[if lt IE 9]>
    	<script src="http://html5shiv.googlecode.com/svn/trunk/html5.js"></script>
  	<![endif]-->
  	<script>
		var _hmt = _hmt || [];
		(function() {
		  var hm = document.createElement("script");
		  hm.src = "//hm.baidu.com/hm.js?94be4b0f9fc5d94cc0d0415ea6761ae9";
		  var s = document.getElementsByTagName("script")[0]; 
		  s.parentNode.insertBefore(hm, s);
		})();
	</script>
  	<style type="text/css">
	  	.docs-content{
	  		margin-bottom: 10px;
	  	}
  	</style>
</head>


  <body class="index">

    <header class="navbar navbar-inverse navbar-fixed-top docs-nav" role="banner">
  <div class="container">
    <div class="navbar-header">
      <button class="navbar-toggle" type="button" data-toggle="collapse" data-target=".bs-navbar-collapse">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a href="/" class="navbar-brand">
        <img src="/styles/images/logo.jpg">
      </a>
    </div>
    <nav class="collapse navbar-collapse bs-navbar-collapse" role="navigation">
      <ul class="nav navbar-nav">    
        <li>
          <a href="/">Home</a>
        </li>
        <li>
          <a href="/categories/">分类</a>
        </li>
        <li>
          <a href="/tag">标签</a>
        </li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
          <a href="/donate/"><strong>打赏</strong></a>
        </li>
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown">关于<b class="caret"></b></a>
          <ul class="dropdown-menu">
            <li><a rel="nofollow" target="_blank" href="https://github.com/liuqibao566">Github</a></li>
            <li><a rel="nofollow" target="_blank" href="">关于作者</a></li>
            <li><a rel="nofollow" href="/books">书架</a></li>
            <li><a rel="nofollow" href="/reference">相关链接</a></li>
            <li><a href="/feed.xml">RSS订阅</a></li>
            <li class="divider"></li>
            <li><a rel="nofollow" target="_blank" href="https://github.com/liuqb566/">本项目</a></li>
          </ul>
        </li>
      </ul>
    </nav>
  </div>
</header>

    <div class="docs-header" id="content">
  <div class="container">
  	
  		<!--
		    <h1>## vcftools 的简单使用</h1>
		    <p>Post on Apr 24, 2017 by <a href="/about">Gossie</a></p>
		-->
		    <h1></h1>
    
  </div>
</div>
    
      
<div class="banner">
  <div class="container">
  	
    	<a href="/categories/#bioinformatics-ref">bioinformatics</a>	/
    	<a href="/tag/#使用手册-ref">使用手册</a>
    
  </div>
</div>

    

    <div class="container docs-container">
  <div class="row">
    <div class="col-md-3">
      <div class="sidebar hidden-print" role="complementary">
        <div id="navigation">
  <h1>目录</h1>
  <ul class="nav sidenav">
<!--
    
      
      
      
      

      
        <li><a href="#year_2017">2017</a>
          <ul class="nav">
            <li><a href="#month_2017_May">May</a></li>
      

      
            
          
              <li><a href="#month_2017_April">April</a></li>
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
              <li><a href="#month_2017_March">March</a></li>
          
        
      
    
      
      
      
      

      

      
            
          
              <li><a href="#month_2017_February">February</a></li>
          
        
      
    
      
      
      
      

      

      
            
          
              <li><a href="#month_2017_January">January</a></li>
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
        
            </ul>
          </li>
          <li><a href="#year_2016">2016</a>
            <ul class="nav">
              <li><a href="#month_2016_December">December</a></li>
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
              <li><a href="#month_2016_November">November</a></li>
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            
          
        
      
    
      
      
      
      

      

      
            </ul>
          </li>
      
    
-->
  </ul>
</div> 
      </div>
    </div>
    <div class="col-md-9" role="main">
      <div class="panel docs-content">
        <div class="wrapper">
            <header class="post-header">
              <h1 class="post-title">## vcftools 的简单使用</h1>
              <!--
                <p class="post-meta">Apr 24, 2017</p>
              -->
              <div class="meta">Posted on <span class="postdate">Apr 24, 2017</span> By <a target="_blank" href="http://liuqibao.me">Gossie</a></div>
              <br />
            </header>
            <article class="post-content">
              <ul id="markdown-toc">
  <li><a href="#section" id="markdown-toc-section">1. 安装</a></li>
  <li><a href="#section-1" id="markdown-toc-section-1">2. 使用</a>    <ul>
      <li><a href="#section-2" id="markdown-toc-section-2">2.2. 位点过滤参数：</a></li>
      <li><a href="#section-3" id="markdown-toc-section-3">2.3. 样本过滤参数</a></li>
      <li><a href="#section-4" id="markdown-toc-section-4">2.4. 基因型过滤参数</a></li>
      <li><a href="#section-5" id="markdown-toc-section-5">2.5. 输出参数</a></li>
      <li><a href="#section-6" id="markdown-toc-section-6">2.6. 比较参数</a></li>
    </ul>
  </li>
</ul>

<p>vcftools 是用于 VCF 和 BCF 格式的遗传变异数据的工具包。
它的功能主要有数据统计、运行数据计算、数据过滤、数据转换。</p>

<h3 id="section">1. 安装</h3>
<div class="highlighter-rouge">git clone https://github.com/vcftools/vcftools.git
cd vcftools
./autogen.sh
./configure
make
make install
</div>
<p>如果是非 root 用户，可以使用：`./configure prefix=’Your path’</p>

<h3 id="section-1">2. 使用</h3>

<div class="highlighter-rouge">vcftools [ --vcf FILE | --gzvcf FILE | --bcf FILE] [ --out OUTPUT PREFIX ] [ FILTERING OPTIONS ] [ OUTPUT OPTIONS ]
</div>
<p>#### 2.1. 基本参数：
- 输入文件参数：
–vcf <输入文件>：输入文件可以是 VCF 格式 v4.0、v4.1 和 v4.2。’-‘ 用于标准输入
--gzvcf <输入文件>： 读入 gzipped 压缩文件
--bcf <输入文件>：读入 BCF2 文件。
- 输出文件参数：
--out <输出文件>：所有输出文件的文件名前缀。如果忽略，默认为 ’out.'
--stdout
-c：输出为标准输出
-temp <temp_文件>：将 vcftools 产生的所有临时文件输入到指定目录。</temp_文件></输出文件></输入文件></输入文件></输入文件></p>

<h4 id="section-2">2.2. 位点过滤参数：</h4>

<ul>
  <li>位置过滤
–chr <chromosome> 
--not-chr <chromosome></chromosome></chromosome></li>
</ul>

<p>匹配或者不匹配某条染色体。可能多次使用该参数，以设定多条染色体。</p>

<p>–from-bp <integer> 
--to-bp <integer></integer></integer></p>

<p>设定碱基的位置范围，仅与 <code class="highlighter-rouge">--chr</code> 结合使用。可以只使用一个。</p>

<p>–positions <filename> 
--exclude-positions <filename></filename></filename></p>

<p>使用或者不使用文件中的某些位置的碱基。输入文件的每行用 <code class="highlighter-rouge">\t</code> 制表符分隔，允许加注释行，用 ‘#’ 开头。</p>

<p>–positions-overlap <filename> 
--exclude-positions-overlap <filename></filename></filename></p>

<p>排除或者不排除文件中与参考等位基因重叠的碱基。</p>

<p>–bed <filename> 
--exclude-bed <filename></filename></filename></p>

<p>Include or exclude a set of sites on the basis of a BED file. Only the first three columns (chrom, chromStart and chromEnd) are required. The BED file is expected to have a header line. A site will be kept or excluded if any part of any allele (REF or ALT) at a site is within the range of one of the BED entries.</p>

<p>–thin <integer></integer></p>

<p>Thin sites so that no two sites are within the specified distance from one another.</p>

<p>–mask <filename> 
--invert-mask <filename> 
--mask-min <integer></integer></filename></filename></p>

<p>These options are used to specify a FASTA-like mask file to filter with. The mask file contains a sequence of integer digits (between 0 and 9) for each position on a chromosome that specify if a site at that position should be filtered or not. 
An example mask file would look like:
<code class="highlighter-rouge">
&gt;1 
0000011111222... 
&gt;2 
2222211111000...
</code>
In this example, sites in the VCF file located within the first 5 bases of the start of chromosome 1 would be kept, whereas sites at position 6 onwards would be filtered out. And sites after the 11th position on chromosome 2 would be filtered out as well. 
The “–invert-mask” option takes the same format mask file as the “–mask” option, however it inverts the mask file before filtering with it. 
And the “–mask-min” option specifies a threshold mask value between 0 and 9 to filter positions by. The default threshold is 0, meaning only sites with that value or lower will be kept.</p>

<ul>
  <li>位点过滤</li>
</ul>

<p>–snp <string></string></p>

<p>包含一个 SNP 位点，如 dbSNP 的 rsID。可以使用多次。</p>

<p>–snps <filename> 
--exclude <filename></filename></filename></p>

<p>使用或者不使用文件中给出的 SNPs。文件包含一列 SNP IDs （如 dbSNP rsIDs），每行一个，不需要表头。</p>

<ul>
  <li>变异类型过滤
–keep-only-indels 
–remove-indels</li>
</ul>

<p>使用或者不使用一个 indel。indel 表示任何与参考等位基因长度不同的变异。</p>

<ul>
  <li>FILTER flag 过滤
–remove-filtered-all</li>
</ul>

<p>根据 vcf 文件中的 FILTER flag 过滤，除了 PASS。</p>

<p>–keep-filtered <string> 
--remove-filtered <string></string></string></p>

<p>使用或者不使用 FILTER flag 标记的所有位点。可以多次使用该参数。</p>

<ul>
  <li>根据 INFO 过滤</li>
</ul>

<p>–keep-INFO <string> 
--remove-INFO <string></string></string></p>

<ul>
  <li>根据 ALLELE 过虑</li>
</ul>

<p>–maf <float> ：过滤大于等于该 MAF 的位点
--max-maf <float>：过滤小于等于该 MAF 的位点。</float></float></p>

<p>–non-ref-af <float> 
--max-non-ref-af <float> 
--non-ref-ac <integer> 
--max-non-ref-ac <integer></integer></integer></float></float></p>

<p>–non-ref-af-any <float> 
--max-non-ref-af-any <float> 
--non-ref-ac-any <integer> 
--max-non-ref-ac-any <integer></integer></integer></float></float></p>

<p>–mac <integer> ：过滤最小等位基因 count 数大于等位该参数的位点
--max-mac <integer>：过滤最小等位基因 count 数小于等于该参数的位点</integer></integer></p>

<p>–min-alleles <integer> 
--max-alleles <integer></integer></integer></p>

<p>过滤等位基因数小于或者大于该参数的位点，可以结合使用，例如，<code class="highlighter-rouge">vcftools --vcf file1.vcf --min-alleles 2 --max-alleles 2</code> 过滤只有两个等位基因的位点。</p>

<ul>
  <li>根据基因型数值过滤</li>
</ul>

<p>–min-meanDP <float> 
--max-meanDP <float></float></float></p>

<p>根据测序深度过滤</p>

<p>–hwe <float>：哈迪-温伯格平衡</float></p>

<p>Assesses sites for Hardy-Weinberg Equilibrium using an exact test, as defined by Wigginton, Cutler and Abecasis (2005). Sites with a p-value below the threshold defined by this option are taken to be out of HWE, and therefore excluded.</p>

<p>–max-missing <float>：根据缺失数据比例过滤</float></p>

<p>Exclude sites on the basis of the proportion of missing data (defined to be between 0 and 1, where 0 allows sites that are completely missing and 1 indicates no missing data allowed).</p>

<p>–max-missing-count <integer></integer></p>

<p>Exclude sites with more than this number of missing genotypes over all individuals.</p>

<p>–phased</p>

<p>Excludes all sites that contain unphased genotypes.</p>

<ul>
  <li>杂项过滤
–minQ <float>：根据 Quality 值过滤</float></li>
</ul>

<h4 id="section-3">2.3. 样本过滤参数</h4>

<p>–indv <string> 
--remove-indv <string></string></string></p>

<p>使用或者不使用某样本，可以多次使用该参数。</p>

<p>–keep <filename> 
--remove <filename></filename></filename></p>

<p>使用或者移除的文件中所列的本亲。样本 ID （与 VCF 表头相同）每个一行。如果同时使用两参数，<code class="highlighter-rouge">--keep</code> 先于 <code class="highlighter-rouge">--remove</code> 运行。如果使用了多个文件，先保留所有 ’–keep‘ 文件中的样本，再删除其中 ’–remove‘ 所有文件中的样本。</p>

<p>–max-indv <integer>: Randomly thins individuals so that only the specified number are retained.</integer></p>

<h4 id="section-4">2.4. 基因型过滤参数</h4>

<p>–remove-filtered-geno-all</p>

<p>Excludes all genotypes with a FILTER flag not equal to “.” (a missing value) or PASS.</p>

<p>–remove-filtered-geno <string></string></p>

<p>Excludes genotypes with a specific FILTER flag.</p>

<p>–minGQ <float></float></p>

<p>Exclude all genotypes with a quality below the threshold specified. This option requires that the “GQ” FORMAT tag is specified for all sites.</p>

<p>–minDP <float> 
--maxDP <float></float></float></p>

<p>Includes only genotypes greater than or equal to the “–minDP” value and less than or equal to the “–maxDP” value. This option requires that the “DP” FORMAT tag is specified for all sites.</p>

<h4 id="section-5">2.5. 输出参数</h4>
<ul>
  <li>输出等位基因统计
-freq 
–freq2</li>
</ul>

<p>输出带有后缀 ’.frq‘ 文件中的每个位点的等位基因频率。第二选项禁止所有关于等位基因信息的输出。</p>

<p>–counts 
–counts2</p>

<p>输出带有后缀 ‘.frq.count’ 文件中的原始等位基因 counts 数。第二个选项禁止所有关于等位基因信息的输出。</p>

<p>–derived</p>

<p>For use with the previous four frequency and count options only. Re-orders the output file columns so that the ancestral allele appears first. This option relies on the ancestral allele being specified in the VCF file using the AA tag in the INFO field.</p>

<ul>
  <li>输出深度统计</li>
</ul>

<p>–depth</p>

<p>Generates a file containing the mean depth per individual. This file has the suffix “.idepth”.</p>

<p>–site-depth</p>

<p>Generates a file containing the depth per site summed across all individuals. This output file has the suffix “.ldepth”.</p>

<p>–site-mean-depth</p>

<p>Generates a file containing the mean depth per site averaged across all individuals. This output file has the suffix “.ldepth.mean”.</p>

<p>–geno-depth</p>

<p>Generates a (possibly very large) file containing the depth for each genotype in the VCF file. Missing entries are given the value -1. The file has the suffix “.gdepth”.</p>

<ul>
  <li>输出 LD 统计</li>
</ul>

<p>–hap-r2</p>

<p>Outputs a file reporting the r2, D, and D’ statistics using phased haplotypes. These are the traditional measures of LD often reported in the population genetics literature. The output file has the suffix “.hap.ld”. This option assumes that the VCF input file has phased haplotypes.</p>

<p>–geno-r2</p>

<p>Calculates the squared correlation coefficient between genotypes encoded as 0, 1 and 2 to represent the number of non-reference alleles in each individual. This is the same as the LD measure reported by PLINK. The D and D’ statistics are only available for phased genotypes. The output file has the suffix “.geno.ld”.</p>

<p>–geno-chisq</p>

<p>If your data contains sites with more than two alleles, then this option can be used to test for genotype independence via the chi-squared statistic. The output file has the suffix “.geno.chisq”.</p>

<p>–hap-r2-positions <positions list="" file=""> 
--geno-r2-positions <positions list="" file=""></positions></positions></p>

<p>Outputs a file reporting the r2 statistics of the sites contained in the provided file verses all other sites. The output files have the suffix “.list.hap.ld” or “.list.geno.ld”, depending on which option is used.</p>

<p>–ld-window <integer></integer></p>

<p>This optional parameter defines the maximum number of SNPs between the SNPs being tested for LD in the “–hap-r2”, “–geno-r2”, and “–geno-chisq” functions.</p>

<p>–ld-window-bp <integer></integer></p>

<p>This optional parameter defines the maximum number of physical bases between the SNPs being tested for LD in the “–hap-r2”, “–geno-r2”, and “–geno-chisq” functions.</p>

<p>–ld-window-min <integer></integer></p>

<p>This optional parameter defines the minimum number of SNPs between the SNPs being tested for LD in the “–hap-r2”, “–geno-r2”, and “–geno-chisq” functions.</p>

<p>–ld-window-bp-min <integer></integer></p>

<p>This optional parameter defines the minimum number of physical bases between the SNPs being tested for LD in the “–hap-r2”, “–geno-r2”, and “–geno-chisq” functions.</p>

<p>–min-r2 <float></float></p>

<p>This optional parameter sets a minimum value for r2, below which the LD statistic is not reported by the “–hap-r2”, “–geno-r2”, and “–geno-chisq” functions.</p>

<p>–interchrom-hap-r2 
–interchrom-geno-r2</p>

<p>Outputs a file reporting the r2 statistics for sites on different chromosomes. The output files have the suffix “.interchrom.hap.ld” or “.interchrom.geno.ld”, depending on the option used.</p>

<ul>
  <li>OUTPUT TRANSITION/TRANSVERSION STATISTICS</li>
</ul>

<p>–TsTv <integer></integer></p>

<p>Calculates the Transition / Transversion ratio in bins of size defined by this option. Only uses bi-allelic SNPs. The resulting output file has the suffix “.TsTv”.</p>

<p>–TsTv-summary</p>

<p>Calculates a simple summary of all Transitions and Transversions. The output file has the suffix “.TsTv.summary”.</p>

<p>–TsTv-by-count</p>

<p>Calculates the Transition / Transversion ratio as a function of alternative allele count. Only uses bi-allelic SNPs. The resulting output file has the suffix “.TsTv.count”.</p>

<p>–TsTv-by-qual</p>

<p>Calculates the Transition / Transversion ratio as a function of SNP quality threshold. Only uses bi-allelic SNPs. The resulting output file has the suffix “.TsTv.qual”.</p>

<p>–FILTER-summary</p>

<p>Generates a summary of the number of SNPs and Ts/Tv ratio for each FILTER category. The output file has the suffix “.FILTER.summary”.</p>

<ul>
  <li>输出核酸多态性统计</li>
</ul>

<p>–site-pi</p>

<p>Measures nucleotide divergency on a per-site basis. The output file has the suffix “.sites.pi”.</p>

<p>–window-pi <integer> 
--window-pi-step <integer></integer></integer></p>

<p>Measures the nucleotide diversity in windows, with the number provided as the window size. The output file has the suffix “.windowed.pi”. The latter is an optional argument used to specify the step size in between windows.</p>

<ul>
  <li>输出 FST 统计</li>
</ul>

<p>–weir-fst-pop <filename></filename></p>

<p>This option is used to calculate an Fst estimate from Weir and Cockerham’s 1984 paper. This is the preferred calculation of Fst. The provided file must contain a list of individuals (one individual per line) from the VCF file that correspond to one population. This option can be used multiple times to calculate Fst for more than two populations. These files will also be included as “–keep” options. By default, calculations are done on a per-site basis. The output file has the suffix “.weir.fst”.</p>

<p>–fst-window-size <integer> 
--fst-window-step <integer></integer></integer></p>

<p>These options can be used with “–weir-fst-pop” to do the Fst calculations on a windowed basis instead of a per-site basis. These arguments specify the desired window size and the desired step size between windows.</p>

<ul>
  <li>输出其它统计</li>
</ul>

<p>–het</p>

<p>Calculates a measure of heterozygosity on a per-individual basis. Specfically, the inbreeding coefficient, F, is estimated for each individual using a method of moments. The resulting file has the suffix “.het”.</p>

<p>–hardy</p>

<p>Reports a p-value for each site from a Hardy-Weinberg Equilibrium test (as defined by Wigginton, Cutler and Abecasis (2005)). The resulting file (with suffix “.hwe”) also contains the Observed numbers of Homozygotes and Heterozygotes and the corresponding Expected numbers under HWE.</p>

<p>–TajimaD <integer></integer></p>

<p>Outputs Tajima’s D statistic in bins with size of the specified number. The output file has the suffix “.Tajima.D”.</p>

<p>–indv-freq-burden</p>

<p>This option calculates the number of variants within each individual of a specific frequency. The resulting file has the suffix “.ifreqburden”.</p>

<p>–LROH</p>

<p>This option will identify and output Long Runs of Homozygosity. The output file has the suffix “.LROH”. This function is experimental, and will use a lot of memory if applied to large datasets.</p>

<p>–relatedness</p>

<p>This option is used to calculate and output a relatedness statistic based on the method of Yang et al, Nature Genetics 2010 (doi:10.1038/ng.608). Specifically, calculate the unadjusted Ajk statistic. Expectation of Ajk is zero for individuals within a populations, and one for an individual with themselves. The output file has the suffix “.relatedness”.</p>

<p>–relatedness2</p>

<p>This option is used to calculate and output a relatedness statistic based on the method of Manichaikul et al., BIOINFORMATICS 2010 (doi:10.1093/bioinformatics/btq559). The output file has the suffix “.relatedness2”.</p>

<p>–site-quality</p>

<p>Generates a file containing the per-site SNP quality, as found in the QUAL column of the VCF file. This file has the suffix “.lqual”.</p>

<p>–missing-indv</p>

<p>Generates a file reporting the missingness on a per-individual basis. The file has the suffix “.imiss”.</p>

<p>–missing-site</p>

<p>Generates a file reporting the missingness on a per-site basis. The file has the suffix “.lmiss”.</p>

<p>–SNPdensity <integer></integer></p>

<p>Calculates the number and density of SNPs in bins of size defined by this option. The resulting output file has the suffix “.snpden”.</p>

<p>–kept-sites</p>

<p>Creates a file listing all sites that have been kept after filtering. The file has the suffix “.kept.sites”.</p>

<p>–removed-sites</p>

<p>Creates a file listing all sites that have been removed after filtering. The file has the suffix “.removed.sites”.</p>

<p>–singletons</p>

<p>This option will generate a file detailing the location of singletons, and the individual they occur in. The file reports both true singletons, and private doubletons (i.e. SNPs where the minor allele only occurs in a single individual and that individual is homozygotic for that allele). The output file has the suffix “.singletons”.</p>

<p>–hist-indel-len</p>

<p>This option will generate a histogram file of the length of all indels (including SNPs). It shows both the count and the percentage of all indels for indel lengths that occur at least once in the input file. SNPs are considered indels with length zero. The output file has the suffix “.indel.hist”.</p>

<p>–hapcount <BED file=""></BED></p>

<p>This option will output the number of unique haplotypes within user specified bins, as defined by the BED file. The output file has the suffix “.hapcount”.</p>

<p>–mendel <PED file=""></PED></p>

<p>This option is use to report mendel errors identified in trios. The command requires a PLINK-style PED file, with the first four columns specifying a family ID, the child ID, the father ID, and the mother ID. The output of this command has the suffix “.mendel”.</p>

<p>–extract-FORMAT-info <string></string></p>

<p>Extract information from the genotype fields in the VCF file relating to a specfied FORMAT identifier. The resulting output file has the suffix “.<FORMAT_ID>.FORMAT". For example, the following command would extract the all of the GT (i.e. Genotype) entries:</FORMAT_ID></p>

<p>vcftools –vcf file1.vcf –extract-FORMAT-info GT</p>

<p>–get-INFO <string></string></p>

<p>This option is used to extract information from the INFO field in the VCF file. The <string> argument specifies the INFO tag to be extracted, and the option can be used multiple times in order to extract multiple INFO entries. The resulting file, with suffix ".INFO", contains the required INFO information in a tab-separated table. For example, to extract the NS and DB flags, one would use the command:</string></p>

<p>vcftools –vcf file1.vcf –get-INFO NS –get-INFO DB</p>

<ul>
  <li>输出 VCF 格式</li>
</ul>

<p>–recode 
–recode-bcf</p>

<p>These options are used to generate a new file in either VCF or BCF from the input VCF or BCF file after applying the filtering options specified by the user. The output file has the suffix “.recode.vcf” or “.recode.bcf”. By default, the INFO fields are removed from the output file, as the INFO values may be invalidated by the recoding (e.g. the total depth may need to be recalculated if individuals are removed). This behavior may be overriden by the following options. By default, BCF files are written out as BGZF compressed files.</p>

<p>–recode-INFO <string> 
--recode-INFO-all</string></p>

<p>These options can be used with the above recode options to define an INFO key name to keep in the output file. This option can be used multiple times to keep more of the INFO fields. The second option is used to keep all INFO values in the original file.</p>

<p>–contigs <string></string></p>

<p>This option can be used in conjuction with the –recode-bcf when the input file does not have any contig declarations. This option expects a file name with one contig header per line. These lines are included in the output file.</p>

<ul>
  <li>输出其它格式</li>
</ul>

<p>–012</p>

<p>This option outputs the genotypes as a large matrix. Three files are produced. The first, with suffix “.012”, contains the genotypes of each individual on a separate line. Genotypes are represented as 0, 1 and 2, where the number represent that number of non-reference alleles. Missing genotypes are represented by -1. The second file, with suffix “.012.indv” details the individuals included in the main file. The third file, with suffix “.012.pos” details the site locations included in the main file.</p>

<p>–IMPUTE</p>

<p>This option outputs phased haplotypes in IMPUTE reference-panel format. As IMPUTE requires phased data, using this option also implies –phased. Unphased individuals and genotypes are therefore excluded. Only bi-allelic sites are included in the output. Using this option generates three files. The IMPUTE haplotype file has the suffix “.impute.hap”, and the IMPUTE legend file has the suffix “.impute.hap.legend”. The third file, with suffix “.impute.hap.indv”, details the individuals included in the haplotype file, although this file is not needed by IMPUTE.</p>

<p>–ldhat 
–ldhat-geno</p>

<p>These options output data in LDhat format. This option requires the “–chr” filter option to also be used. The first option outputs phased data only, and therefore also implies “–phased” be used, leading to unphased individuals and genotypes being excluded. The second option treats all of the data as unphased, and therefore outputs LDhat files in genotype/unphased format. Two output files are generated with the suffixes “.ldhat.sites” and “.ldhat.locs”, which correspond to the LDhat “sites” and “locs” input files respectively.</p>

<p>–BEAGLE-GL 
–BEAGLE-PL</p>

<p>These options output genotype likelihood information for input into the BEAGLE program. The VCF file is required to contain FORMAT fields with “GL” or “PL” tags, which can generally be output by SNP callers such as the GATK. Use of this option requires a chromosome to be specified via the “–chr” option. The resulting output file has the suffix “.BEAGLE.GL” or “.BEAGLE.PL” and contains genotype likelihoods for biallelic sites. This file is suitable for input into BEAGLE via the “like=” argument.</p>

<p>–plink 
–plink-tped 
–chrom-map</p>

<p>这些参数输出 PLINK PED 格式，产生两个文件，’.ped’ 和 ‘.map‘。仅双等位基因位点会被输出。具体请看 PLINK 说明文件。
注意：第一个选项对大数据集很慢。用户可以作用 ’–chr‘ 参数分割文件，或者使用 ‘–plink-tped’ 参数输出 PLINK 转置格式的文件，后缀为 ‘.tped’ 和 ‘.tfam’。
For usage with variant sites in species other than humans, the –chrom-map option may be used to specify a file name that has a tab-delimited mapping of chromosome name to a desired integer value with one line per chromosome. This file must contain a mapping for every chromosome value found in the file.</p>

<h4 id="section-6">2.6. 比较参数</h4>

<p>These options are used to compare the original variant file to another variant file and output the results. All of the diff functions require both files to contain the same chromosomes and that the files be sorted in the same order. If one of the files contains chromosomes that the other file does not, use the –not-chr filter to remove them from the analysis.</p>

<p>DIFF VCF FILE</p>

<p>–diff <filename> 
--gzdiff <filename> 
--diff-bcf <filename></filename></filename></filename></p>

<p>These options compare the original input file to this specified VCF, gzipped VCF, or BCF file. These options must be specified with one additional option described below in order to specify what type of comparison is to be performed. See the examples section for typical usage.</p>

<p>DIFF OPTIONS</p>

<p>–diff-site</p>

<p>Outputs the sites that are common / unique to each file. The output file has the suffix “.diff.sites_in_files”.</p>

<p>–diff-indv</p>

<p>Outputs the individuals that are common / unique to each file. The output file has the suffix “.diff.indv_in_files”.</p>

<p>–diff-site-discordance</p>

<p>This option calculates discordance on a site by site basis. The resulting output file has the suffix “.diff.sites”.</p>

<p>–diff-indv-discordance</p>

<p>This option calculates discordance on a per-individual basis. The resulting output file has the suffix “.diff.indv”.</p>

<p>–diff-indv-map <filename></filename></p>

<p>This option allows the user to specify a mapping of individual IDs in the second file to those in the first file. The program expects the file to contain a tab-delimited line containing an individual’s name in file one followed by that same individual’s name in file two with one mapping per line.</p>

<p>–diff-discordance-matrix</p>

<p>This option calculates a discordance matrix. This option only works with bi-allelic loci with matching alleles that are present in both files. The resulting output file has the suffix “.diff.discordance.matrix”.</p>

<p>–diff-switch-error</p>

<p>This option calculates phasing errors (specifically “switch errors”). This option creates an output file describing switch errors found between sites, with suffix “.diff.switch”.</p>

            </article>
        </div>
      </div>
      <div class="panel docs-content">
        <article class="post-content">
          <div class="wrapper">
            


  <div class="ds-thread" data-thread-key="/2017/04/24/vcftools-introduction/" data-title="## vcftools 的简单使用" data-url="http://liuqibao.me/2017/04/24/vcftools-introduction/"></div>

<script type="text/javascript">
var duoshuoQuery = {short_name:"Gossie"};
	(function() {
		var ds = document.createElement('script');
		ds.type = 'text/javascript';ds.async = true;
		ds.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//static.duoshuo.com/embed.js';
		ds.charset = 'UTF-8';
		(document.getElementsByTagName('head')[0] 
		 || document.getElementsByTagName('body')[0]).appendChild(ds);
	})();
</script>


 
          </div>
        </article>
      </div>
    </div>
  </div>
</div>

    
    <footer class="footer" role="contentinfo">
	<div class="container">
		<p class="copyright">Copyright &copy; 2014-2017 <a href=""><code>Gossie</code></a>.</p>
		<p>Site powered by <a href="http://jekyllrb.com">Jekyll</a> && <a href="http://www.github.com">Github</a></p>
	</div>
</footer>

<script src="/styles/js/jquery.min.js"></script>
<script src="/styles/js/bootstrap.min.js"></script>
<script src="/styles/js/holder.min.js"></script>
<script src="/styles/js/application.js"></script>
<script src="/styles/js/lessismore.js"></script>


  </body>
</html>
